oracle/config/market.go

```diff
@@ -0,0 +1,102 @@
+package config
+
+import (
+	"fmt"
+
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// MarketConfig represents the provider specific configurations for different
+// currency pairs and the corresponding markets they are traded on.
+type MarketConfig struct {
+	// Name identifies which provider this config is for.
+	Name string `mapstructure:"name" toml:"name"`
+
+	// CurrencyPairToMarketConfigs is the config the provider uses to create mappings
+	// between on-chain and off-chain currency pairs. In particular, this config
+	// maps the on-chain currency pair representation (i.e. BITCOIN/USD) to the
+	// off-chain currency pair representation (i.e. BTC/USD).
+	CurrencyPairToMarketConfigs map[string]CurrencyPairMarketConfig `mapstructure:"currency_pair_to_market_configs" toml:"currency_pair_to_market_configs"`
+}
+
+// CurrencyPairMarketConfig is the config the provider uses to create mappings
+// between on-chain and off-chain currency pairs.
+type CurrencyPairMarketConfig struct {
+	// Ticker is the ticker symbol for the currency pair off-chain.
+	Ticker string `mapstructure:"ticker" toml:"ticker"`
+
+	// CurrencyPair is the on-chain representation of the currency pair.
+	CurrencyPair oracletypes.CurrencyPair `mapstructure:"currency_pair" toml:"currency_pair"`
+}
```

Comment:
Why can we not just do a map of Ticker -> CurrencyPair ?

Reply:
toml has some strange behavior with the keys of maps (it always lower cases them), so you might be losing some amount of information when you use the ticker as the key. IMO, this also provides better standardization and if you take a look at what the dydx oracle config looks like you'll see that it is very similar.

Reply:
turns out it was an issue on my end, gonna update this.